pkg/common/commands/resetoffset/controller/offsets.go

```diff
@@ -88,104 +88,117 @@ func (r *Reconciler) reconcileOffsets(ctx context.Context, topicName string, gro
 		return nil, err
 	}
 
-	// Track Partition Specific Resources
-	partitionOffsetManagers := make([]sarama.PartitionOffsetManager, len(partitions))
+	// Create The Required PartitionOffsetManagers For The Specified Topic / Partitions
+	partitionOffsetManagers, err := createPartitionOffsetManagers(offsetManager, topicName, partitions)
+	if err != nil {
+		logger.Error("Failed to create PartitionOffsetManagers for Topic Partitions", zap.Error(err))
+		_ = closeManagersAndDrainErrors(logger, offsetManager, partitionOffsetManagers)
+		return nil, err
+	}
+
+	// Update All Topic Partitions To The Specified Offset Time
+	offsetMappings, err := updateOffsets(logger, saramaClient, offsetManager, partitionOffsetManagers, topicName, partitions, offsetTime)
+	if err != nil {
+		logger.Error("Failed to update Offsets for Topic Partitions", zap.Error(err))
+		_ = closeManagersAndDrainErrors(logger, offsetManager, partitionOffsetManagers)
+		return nil, err
+	}
+
+	// Close The Sarama Managers And Get Any Accumulated Errors
+	err = closeManagersAndDrainErrors(logger, offsetManager, partitionOffsetManagers)
+	if err != nil {
+		logger.Error("")
```

Comment:
I assume this is not intentionally left blank

Reply:
doh!